core/src/main/java/org/apache/lucene/search/vectorhighlight/CustomFieldQuery.java

```diff
@@ -75,6 +82,10 @@ void flatten(Query sourceQuery, IndexReader reader, Collection<Query> flatQuerie
         } else if (sourceQuery instanceof ToParentBlockJoinQuery) {
             ToParentBlockJoinQuery blockJoinQuery = (ToParentBlockJoinQuery) sourceQuery;
             flatten(blockJoinQuery.getChildQuery(), reader, flatQueries, boost);
+        } else if (sourceQuery instanceof BoostingQuery) {
+            BoostingQuery boostingQuery = (BoostingQuery) sourceQuery;
+            flatten(boostingQuery.getContext(), reader, flatQueries, boostingQuery.getBoost());
```

Comment:
Maybe this should be `boost * boostingQuery.getBoost()`? I think that is closer to the way that the boosts are applied.